Zend/tests/gh17162.phpt

```diff
@@ -0,0 +1,22 @@
+--TEST--
+GH-17162 (zend_array_try_init() with dtor can cause engine UAF)
+--FILE--
+<?php
+class Test {
+    function __destruct() {
+        global $box;
+        $box->value = null;
+    }
+}
+$box = new stdClass();
```

Comment:
Nit: I don't think this `stdClass` is needed? Technically it should be an array, should have the same result. This doesn't fail because `$image_info` only has a doc-block type.